yt_geotiff/data_structures.py

```diff
@@ -185,12 +186,53 @@ def _count_grids(self):
         self.num_grids = 1
 
 
+class JPEG2000Hierarchy(GeoTiffHierarchy):
+    grid = GeoTiffGrid   
+
+    def _detect_output_fields(self):
+        # check data dir of the given jp2 file and grab all similarly named files.
+
+        # List of band files in s2 directory
+        s2_band_file_list = [os.path.basename(x) \
+         for x in glob.glob(self.ds.directory+'/*_***_***.jp2')]
+
+        # Follow example for GeoTiffHierarchy to populate the field list.
+        self.field_list = []
+        self.ds.field_units = self.ds.field_units or {}
+
+        # Filename dictionary
+        self.ds._field_filename = {}
+
+        # extract s2 band name from file name
+        def get_band_name(band_file_list):
+            band_file = band_file_list.split("_")
+            return (band_file)
+
+        band_names = [list(b) for b in zip(*map(get_band_name, s2_band_file_list))][2]
+
+        # Attribute with resolution of original loaded image
+        #self.ds._load_resolution = self.ds.resolution.d[0]
+
+        for _i in range(1, len(s2_band_file_list) + 1):
```

Comment:
As far as I can tell, you only ever use `_i-1`, so this would be a bit cleaner as
```suggestion
        for _i in range(len(s2_band_file_list)):
```
and converting `_i-1` to `_i`.